server/repositories/__tests__/prisonApi.spec.js

```diff
@@ -0,0 +1,221 @@
+const { PrisonApiRepository } = require('../prisonApi');
+const { lastCall } = require('../../../test/test-helpers');
+
+describe('PrisonApiRepository', () => {
+  const client = { get: jest.fn() };
+
+  const apiUrl = 'http://foo.bar';
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('getTransactionsFor', () => {
+    it('should return when the transactions request succeeds', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.resolve('API_RESPONSE'));
+
+      const response = await repository.getTransactionsFor(
+        'A123BC',
+        'spends',
+        new Date('2020-01-01'),
+        new Date('2020-01-31'),
+      );
+
+      expect(lastCall(client.get)[0]).toContain(
+        '/offenders/A123BC/transaction-history',
+      );
+      expect(lastCall(client.get)[0]).toContain('account_code=spends');
+      expect(lastCall(client.get)[0]).toContain('from_date=2020-01-01');
+      expect(lastCall(client.get)[0]).toContain('to_date=2020-01-31');
+
+      expect(response).toBe('API_RESPONSE');
+    });
+
+    it('should throw when no prisoner ID is passed', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.resolve('API_RESPONSE'));
+
+      let hasThrown = false;
+
+      try {
+        await repository.getTransactionsFor(
+          null,
+          'spends',
+          new Date('2020-01-01'),
+          new Date('2020-01-31'),
+        );
+      } catch (e) {
+        hasThrown = true;
+        expect(client.get).not.toHaveBeenCalled();
+      }
+
+      expect(hasThrown).toBe(true);
```

Comment:
Quite a few of the other projects use `nock` so they can test the client calls are made correctly. 
There's also pretty good support in jest for async testing

An example for combining these might be:

```js
it('Should log 404 correctly', async () => {
   nock(hostname).get('/api/users/me').reply(404)

   await expect(client.get({}, '/api/users/me')).rejects.toThrow('Not Found')

   expect(logger.warn).toHaveBeenCalledWith('GET /api/users/me No record found')
 })
```

Reply:
I think we went the route of mocking Axios through dependency injection here, however would be nice to eventually include to use something like Nock for mocking our responses and testing this integration. I'll card this one up as it's something I would would prefer to introduce separately than quickly fit in this PR